process/asyncExecution/headersExecutor.go

```diff
@@ -107,6 +115,18 @@ func (he *headersExecutor) process(pair queue.HeaderBodyPair) error {
 		return nil
 	}
 
+	he.blockChain.SetFinalBlockInfo(
+		executionResult.GetHeaderNonce(),
+		executionResult.GetHeaderHash(),
+		executionResult.GetRootHash(),
+	)
+
+	he.blockChain.SetLastExecutedBlockInfo(
```

Comment:
the SetLastExecutedBlockInfo is set only once, here. The SetFinalBlockInfo is set multiple times in multiple places. Is it ok?